pyang/statements.py

```diff
@@ -1785,6 +1785,22 @@ def add_tmp_children(node, tmp_children):
                 node.i_children.append(tmp)
                 tmp.parent = node
 
+    def keyword_is_expected(pos, parent, child):
+        if parent.keyword == '__tmp_augment__':
+            return True
+        if parent.keyword == 'choice' and child.keyword != 'case':
+            (_arg_type, subspec) = grammar.stmt_map['case']
+        else:
+            (_arg_type, subspec) = grammar.stmt_map[parent.keyword]
+        subspec = grammar.flatten_spec(subspec)
+        if util.keysearch(child.keyword, 0, subspec) is None:
+            err_add(ctx.errors, pos, 'UNEXPECTED_KEYWORD_AUGMENT',
+                    (util.keyword_to_str(child.raw_keyword),
+                     util.keyword_to_str(parent.raw_keyword),
+                     child.pos))
```

Comment:
Maybe this statement should be moved out of this function because this function is should be simply just a judgement. Then we can see this function more clear if we do so.

Reply:
Sure. I will update it by your comments.